src/scriptChallenge.js

```diff
@@ -0,0 +1,137 @@
+import loadData from "./js/loadData.js";
+import {
+  filterByTypeOnLoad,
+  getQueryParams,
+  renderChallenges,
+} from "./js/filterByTypeOnLoad.js";
+import { filterAndUpdateCards } from "./js/renderChallenges.js";
+import createChallengeCard from "./js/challengeCard.js";
+import { filterByTypes } from "./js/filterType.js";
+import { getTags, displayTags, filterByTags } from "./js/tagFilter.js";
+import filterText from "./js/filterText.js";
+import { handleStarInput, filterByRating } from "./js/filterRating.js";
+
+const challengeContainer = document.querySelector(".challenge-list");
+const textFilter = document.querySelector("#text-filter");
+const onlineTypeInput = document.querySelector(
+  "input[name='online-challenge']"
+);
+const onsiteTypeInput = document.querySelector(
+  "input[name='on-site-challenge']"
+);
+const typeInputs = [onlineTypeInput, onsiteTypeInput];
+const tagParent = document.querySelector("#btn-container");
+const ratingStars = document.querySelectorAll(".rating-star");
+
+async function run() {
+  const challenges = await loadData();
+  const queryParams = getQueryParams();
+  renderChallenges(
+    queryParams,
+    challengeContainer,
+    challenges,
+    filterByTypeOnLoad,
+    createChallengeCard
+  );
+
+  if(queryParams === "online") {
+    onlineTypeInput.setAttribute("checked", "true");
+  } 
+  else if(queryParams === "onsite") {
+    onsiteTypeInput.setAttribute("checked", "true");
+  } else {
+    onlineTypeInput.setAttribute("checked", "true");
+    onsiteTypeInput.setAttribute("checked", "true");
+  }
+
+  //Add eventlistener for stars
+  handleStarInput();
+  ratingStars.forEach((star)=> {
+    star.addEventListener("click", async () => {
+        let selectedTags = tagParent.querySelectorAll(".active");
+        selectedTags = Array.from(selectedTags);
+        selectedTags = selectedTags.map((tag) => {
+          return tag.textContent;
+        });
+        let data = challenges;
+        console.log(data)
+        data = await filterByTags(selectedTags, data);
+        filterAndUpdateCards(
+          data,
+          createChallengeCard,
+          challengeContainer,
+          filterText,
+          filterByTypes,
+          filterByRating
+        );
+      });
+  });
+
+  const tags = getTags(challenges);
+  const buttons = displayTags(tags, challenges, tagParent);
+  
+  //Add eventlistener for tag buttons
+  (await buttons).forEach((button) => {
```

Comment:
`buttons` shouldn't have to be asynchronous. This to me is an indication of how the async parts of the code have "contaminated" everything, and now all of a sudden even simple lists of buttons have to be async.